tests/accessor/accessor_api_image_common.h

```diff
@@ -427,16 +463,18 @@ T read_image_acc_sampled(const sycl::accessor<T, dims, mode, target> &acc,
     return acc.read(coords, smpl);
   }
 }
-template <typename T, int dims, sycl::access_mode mode, typename coordT>
-T read_image_acc_sampled(const sycl::accessor<T, dims, mode,
+template <typename T, int dims, sycl::access_mode access_mode, typename coordT>
+T read_image_acc_sampled(const sycl::accessor<T, dims, access_mode,
                                     sycl::target::image_array> &acc,
                          sycl::sampler smpl,
                          image_array_id_t<dims> idx,
                          image_array_range_t<dims> range,
-                         const coordT& coordTag) {
+                         const coordT& coordTag,
+                         sycl::filtering_mode filteringMode) {
   // Verify __image_array_slice__ read
   const auto coords = image_array_coords<dims>::get(coordTag, idx, range);
-  return acc[idx[dims]].read(coords, smpl);
+  return acc[get_slice_index<dims>(coordTag, filteringMode, idx)].read(coords,
```

Comment:
Currently the range is defined by using the `image_accessor_sampler::supports_out_of_range` method, so for the linear filtering  mode and image with texels `0 .. N-1` we have verification for `acc[0].read() ... acc[N].read()`.

Please correct me if I'm wrong, but looks like with this patch we are going change verification range from `0 .. N` to `1 .. N+1`, switching off the verification of `acc[0].read()` in case of linear filtration and `acc_coord_tag::use_normalized_upper`, but using two out-of-bound indexes instead.

Is it intended, or may be it's better to tune the `get_expected_value<coordT>(idx)` used on the [L1108](https://github.com/KhronosGroup/SYCL-CTS/pull/235/files#diff-cffc0211756d6bcabcad4ed222118165250a78dc0187548ab21285c84258f2a4R1108)?

Reply:
Hm... It looks like you are right, the test after the change just skips `acc[0].read()`. This is similar how `get_expected_value` returns `get_expected_value_linear(idx + 1)` in case of linear filtration and `acc_coord_tag::use_normalized_upper`. So, if we have `idx` is equal to `0`, we calculate an expected value for `1` and for `target::image` we compare this expected value with the actual value read from `get_normalized(const pixel_tag::upper, ...)` coordinates (the coordinates for `0+1` with some manipulations). For `target::image_array`, the last number in `idx` is the image index, while the others are used to calculate the coordinates to read from. When we did `idx + 1` in the `get_expected_value` method, we incremented all the components of `idx` and we should increment the image index we are reading from to match this increment.

I'm not sure what do you mean when you said about to tune the `get_expected_value<coordT>(idx)`. I think we should increment all the components of `idx` but the last one if we are working with `target::image_array`. I've tried and it works, the code has got fewer changes that I proposed before. Thank you for the suggestion. I committed a new patch, please have a look.

Reply:
@yuriykoch Unfortunately the "Re-request review" button doesn't work. Could you review the latest commit?